akka-persistence/src/main/scala/akka/persistence/Persistent.scala

```diff
@@ -246,3 +251,26 @@ private[persistence] final case class PersistentImpl(
     s"PersistentRepr($persistenceId,$sequenceNr,$writerUuid,$timestamp,$metadata)"
   }
 }
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object CompositeMetadata {
+  def extract[M: ClassTag](metadata: Option[Any]): Option[M] = {
+    val metadataType = implicitly[ClassTag[M]].runtimeClass
+    metadata.flatMap {
+      case CompositeMetadata(entries) =>
+        entries.collectFirst {
```

Comment:
What if I would like to keep 2 strings in the metadata? I thought that CompositeMetadata will be a key-value collection.

Reply:
You'd use a concrete type with two string fields in it, not use a generic/stdlib type directly

Reply:
If key-value then different library code and user code would have to agree on unique keys to not overwrite each other's values accidentally. Easier to use key on the concrete class since that is unique for the specific use of the metadata (as long as it is not a general purpose class).

Reply:
But there is also an implicit agreement though. Each library or user code will have to bring its own type and usage of `String` of common/primitive types will lead to bugs. 

We could work with a Seq[Any] and provide a `find` method instead. On the user to decide what is a match.

Reply:
The user would anyway not be able to distinguish a String from another String in such find.